File: packages/sdk/examples/list-tasks.ts
#!/usr/bin/env node
/**
 * Example: List tasks by queue
 *
 * Usage:
 *   OCTOPOID_SERVER_URL=https://... npm run example:list-tasks
 */

import { OctopoidSDK } from '../src/index'

async function main() {
  const sdk = new OctopoidSDK()

  console.log('📊 Task Summary\n')

  // Get tasks by queue
  const queues = ['incoming', 'claimed', 'provisional', 'done'] as const

  for (const queue of queues) {
    const tasks = await sdk.tasks.list({ queue, limit: 100 })

    console.log(`${queue.toUpperCase()}: ${tasks.length}`)

    if (tasks.length > 0) {
      // Show first 3 tasks
      tasks.slice(0, 3).forEach((task) => {
        console.log(`  • ${task.id} (${task.priority}) - ${task.role || 'no role'}`)
      })

      if (tasks.length > 3) {
        console.log(`  ... and ${tasks.length - 3} more`)
      }
    }

    console.log('')
  }
}

main().catch((error) => {
  console.error('❌ Error:', error.message)
  process.exit(1)
})

File: packages/python-sdk/octopoid_sdk/examples/list_tasks.py
#!/usr/bin/env python3
"""
Example: List tasks by queue

Usage:
    OCTOPOID_SERVER_URL=https://... python list_tasks.py
"""

from octopoid_sdk import OctopoidSDK


def main():
    sdk = OctopoidSDK()

    print("📊 Task Summary\n")

    # Get tasks by queue
    queues = ["incoming", "claimed", "provisional", "done"]

    for queue in queues:
        tasks = sdk.tasks.list(queue=queue, limit=100)

        print(f"{queue.upper()}: {len(tasks)}")

        if tasks:
            # Show first 3 tasks
            for task in tasks[:3]:
                role = task.get("role") or "no role"
                print(f"  • {task['id']} ({task['priority']}) - {role}")

            if len(tasks) > 3:
                print(f"  ... and {len(tasks) - 3} more")

        print()


if __name__ == "__main__":
    try:
        main()
    except Exception as e:
        print(f"❌ Error: {e}")
        exit(1)
